torch/csrc/distributed/rpc/tensorpipe_agent.h

```diff
@@ -79,7 +76,7 @@ struct CpuChannelRegistration {
 C10_DECLARE_REGISTRY(TensorPipeCpuChannelRegistry, CpuChannelRegistration);
 
 struct CudaChannelRegistration {
-#if TENSORPIPE_USE_CUDA_IPC_CHANNEL
+#ifdef USE_CUDA_NOT_ROCM
```

Comment:
I'm wondering if there's a reason you're guarding just the _content_ of this struct with the #ifdef, rather than guarding the definition of the whole struct and the declaration of the corresponding registry. In other words, do we need the registry declaration to exist when CUDA is disabled?

Reply:
if we guard the entire struct, we will need more `ifdef` to guard `C10_DECLARE_REGISTRY`, `C10_DEFINE_REGISTRY`, and where we check channels. I am OK either way, let me know which way you prefer. I can add a followup PR to update that.